Client/Assets/ContextMenu.js

```diff
@@ -77,17 +78,30 @@ function handleTouchMove(event) {
 function handleTouchEnd(event) {
   cancel();
 
-  removeEventListener("touchend", handleTouchEnd);
-  removeEventListener("touchmove", handleTouchMove);
+  window.removeEventListener("touchend", handleTouchEnd);
+  window.removeEventListener("touchmove", handleTouchMove);
+
+  setTimeout(function() {
+    if (element) {
+      element.removeEventListener("click", handleClick, true);
+    }
+  });
 
   // If we're showing the context menu, prevent the page from handling the click event.
   if (touchHandled) {
     touchHandled = false;
     event.preventDefault();
+    event.stopPropagation();
   }
 }
 
-addEventListener("touchstart", function (event) {
+function handleClick(event) {
+  this.removeEventListener("click", handleClick, true);
+  event.preventDefault();
+  event.stopPropagation();
+}
+
+window.addEventListener("touchstart", function(event) {
```

Comment:
Just added `window` here to add clarity to the code. This is effectively a non-change.